### PR TITLE
[ENH] add task entity for anat

### LINKED ISSUE #1564: Implement support for task entity
Related to:

- https://github.com/bids-standard/bids-specification/pull/1185
- https://github.com/bids-standard/bids-examples/issues/341

@melanieganz

### PR DESCRIPTION
**Pull request by Remi-Gau**
closes #1564

@melanieganz 

TODO:

- [x] add task metadata that must be validated if the task entity is present
- [x] test against https://github.com/bids-standard/bids-examples/pull/343

### CODE REVIEW

File: bids-validator/bids_validator/rules/session_level_rules.json [hunk 1]
```diff
@@ -33,14 +33,14 @@
   },
 
   "pet_ses": {
-    "regexp": "^[\\/\\\\](sub-[a-zA-Z0-9]+)[\\/\\\\](?:(ses-[a-zA-Z0-9]+)[\\/\\\\])?\\1(_\\2)?(?:_trc-[a-zA-Z0-9]+)?(?:_rec-[a-zA-Z0-9]+)?(?:_task-[a-zA-Z0-9]+)?(?:_run-[0-9]+_)?(@@@_pet_ses_type_@@@)$",
+    "regexp": "^[\\/\\\\](sub-[a-zA-Z0-9]+)[\\/\\\\](?:(ses-[a-zA-Z0-9]+)[\\/\\\\])?\\1(_\\2)?(?:_task-[a-zA-Z0-9]+)?(?:_trc-[a-zA-Z0-9]+)?(?:_rec-[a-zA-Z0-9]+)?(?:_run-[0-9]+_)?(@@@_pet_ses_type_@@@)$",
```
Comment (Remi-Gau) [inline bids-validator/bids_validator/rules/session_level_rules.json:36]: Correcting a mistake as task entity should come before trc and rec.

File: bids-validator/bids_validator/rules/top_level_rules.json [hunk 2]
```diff
@@ -52,14 +52,14 @@
   },
 
   "pet_top": {
-    "regexp": "^[\\/\\\\](?:ses-[a-zA-Z0-9]+_)?(?:trc-[a-zA-Z0-9]+_)?(?:rec-[a-zA-Z0-9]+_)?(?:task-[a-zA-Z0-9]+_)?(?:run-[0-9]+_)?(@@@_pet_suffixes_@@@)\\.json$",
+    "regexp": "^[\\/\\\\](?:ses-[a-zA-Z0-9]+_)?(?:task-[a-zA-Z0-9]+_)?(?:trc-[a-zA-Z0-9]+_)?(?:rec-[a-zA-Z0-9]+_)?(?:run-[0-9]+_)?(@@@_pet_suffixes_@@@)\\.json$",
```
Comment (Remi-Gau) [inline bids-validator/bids_validator/rules/top_level_rules.json:55]: Correcting a mistake as task entity should come before trc and rec.

**Review by effigies — DISMISSED** [review]

**Comment by effigies** [comment]
@Remi-Gau Are you planning that TODO for this PR or another?

**Comment by Remi-Gau** [comment]
I think those are still TODO: was not able to test things locally without failure on the new example

**Comment by effigies** [comment]
@Remi-Gau To be clear, you're planning on getting to those TODO items in this PR and not a follow-up?

**Comment by Remi-Gau** [comment]
In this PR.
Don't make much sense otherwise.
Feel free to switch this to a draft pr, given the progress I am making.

**Comment by rwblair** [comment]
@Remi-Gau currently anatomical jsons don't have use jsonschema for validation. Since none of this metadata is required and the warnings will be issued in the new schema based valdiator I'm say its fine to skip implementing the metadata checks for this entity in the old validator.

**Comment by Remi-Gau** [comment]
> @Remi-Gau currently anatomical jsons don't have use jsonschema for validation. Since none of this metadata is required and the warnings will be issued in the new schema based valdiator I'm say its fine to skip implementing the metadata checks for this entity in the old validator.

make sense to me

**Pull request merged**